omnibus/config/software/datadog-agent.rb

```diff
@@ -237,4 +240,26 @@
   unless windows_target?
     delete "#{install_dir}/uselessfile"
   end
+
+  # TODO: move this to omnibus-ruby::health-check.rb
+  # check that linux binaries contains OpenSSL symbols when building to support FIPS
+  if fips_mode? && linux_target?
+
+      # Put the ruby code in a block to prevent omnibus from running it directly but rather at build step with the rest of the code above.
+      # If not in a block, it will search for binaries that have not been built yet.
+      block do
+        LINUX_BINARIES = [
+          "#{install_dir}/bin/agent/agent",
+          "#{install_dir}/embedded/bin/trace-agent",
+          "#{install_dir}/embedded/bin/process-agent",
+          "#{install_dir}/embedded/bin/security-agent",
+          "#{install_dir}/embedded/bin/system-probe",
+        ]
+
+        linux_symbol_checker = FIPSComplianceChecker.new("_Cfunc_go_openssl")
```

Comment:
If we decide to stay using the `go tool` for this, we already have a class that handles a similar logic `GoSymbolsInspector`

```ruby
LINUX_BINARIES = [
  "#{install_dir}/bin/agent/agent",
  "#{install_dir}/embedded/bin/trace-agent",
  "#{install_dir}/embedded/bin/process-agent",
  "#{install_dir}/embedded/bin/security-agent",
  "#{install_dir}/embedded/bin/system-probe",
]

symbol = "_Cfunc_go_openssl"

check_block = proc do  |symbols| 
  count = symbols.scan(symbol).count
  if count > 0
    puts "Symbol '#{symbol}' found #{count} times in binary '#{@binary}'."
  else
    puts "Unix: Symbols not found for #{@binary}: '#{symbols}'"
    raise FIPSSymbolsNotFound.new("Expected to find '#{@symbol}' symbol in #{@binary} but did not")
  end
end

LINUX_BINARIES.each do |bin|
  GoSymbolsInspector.new(bin, check_block).inspect()
end
```

Reply:
9ca6c56f6a6ac1d69ff408528f768e42bd18aa45